app/src/main/java/com/asfoundation/wallet/ui/iab/InAppPurchaseInteractor.java

```diff
@@ -202,6 +202,17 @@ private Payment mapToBdsPayment(Payment transaction, Purchase purchase) {
         .firstOrError();
   }
 
+  public Single<Double> getTransactionAmount(String uid) {
```

Comment:
can you remove duplicated code in getTransactionAmount and getTransactionUid methods?